gnocchi/rest/api.py

```diff
@@ -235,6 +240,22 @@ def get_pagination_options(params, default):
 )
 
 
+class UnableToDecodeBody(Exception):
+    def __init__(self, exception, body):
+        self.text_type = six.text_type(exception)
+        self.body = body
+        super(UnableToDecodeBody, self).__init__(body)
+
+    def jsonify(self):
+        return {
+            "cause": "Unable to decode body",
+            "detail": {
+                "text_type": self.text_type,
+                "body": self.body
+            }
+        }
```

Comment:
i would change this to:

```
class UnableToDecodeBody(Exception):
    def __init__(self, exception, body):
        self.reason = six.text_type(exception)
        self.body = body
        super(UnableToDecodeBody, self).__init__("Unable to decode body")

    def jsonify(self):
        return {
            "cause": "Unable to decode body",
            "reason": self.reason,
            "detail": self.body
        }
```
and temporarily put it along side `gnocchi/rest/aggregates/exceptions.py` (or move it to `gnocchi/rest/exceptions.py`)

Reply:
Ok, I extracted this class to a new file 'gnocchi/rest/exceptions.py'